inst/create_cm_template.R

```diff
@@ -0,0 +1,230 @@
+  # Derive CMDECOD
+  assign_no_ct(
+    raw_dataset = MD1,
+    raw_variable = CMDECOD,
+    target_sdtm_var = CMDECOD
+  ) |>
+  # Derive CMSTDTC. This function calls create_iso8601
+  assign_no_ct(
+    raw_dataset = MD1,
+    raw_variable = c("MDBD", "MDBTM"),
+    target_sdtm_var = CMSTDTC,
+    .format = c("ddmmmyyyy", "HHMM")
+  ) |>
+  # Derive CMENDTC. This function calls create_iso8601
+  assign_no_ct(
+    raw_dataset = MD1,
+    raw_variable = c("MDED", "MDETM"),
+    target_sdtm_var = CMENDTC,
+    .format = c("ddmmmyyyy", "HHMM")
+  ) |>
+  # Derive qualifier CMSTRTPT  Annotation text = If checked then CM.CMSTRTPT = 'BEFORE'
+  if_then_else(
+    raw_dataset = MD1,
+    raw_variable = MDPRIOR,
+    condition_left_raw_dataset = MD1,
+    condition_left_raw_variable = MDPRIOR,
+    condition_operator = "if_checked",
+    sub_algorithm = assign_ct,
+    target_sdtm_variable_codelist_code = "C66728",
+    target_sdtm_var = CMSTRTPT,
+    merge_to_topic_by = c(oak_id_vars,
+      topic_var_source = MDRAW
+    )
+  ) |>
+  # Derive qualifier CMSTRTPT  Annotation text = If checked then CM.CMSTTPT = 'SCREENING'
+  if_then_else(
+    raw_dataset = MD1,
+    raw_variable = MDPRIOR,
+    condition_left_raw_dataset = MD1,
+    condition_left_raw_variable = MDPRIOR,
+    condition_operator = "if_checked",
+    sub_algorithm = hardcode_no_ct,
+    target_hardcoded_value = "SCREENING",
+    target_sdtm_var = CM.CMSTTPT,
+    merge_to_topic_by = c(oak_id_vars,
+      topic_var_source = MDRAW
+    )
+  ) |>
+  # Derive qualifier CMSTRTPT  Annotation text = If checked then CM.CMENRTPT = 'ONGOING'
+  if_then_else(
+    raw_dataset = MD1,
+    raw_variable = MDONG,
+    condition_left_raw_dataset = MD1,
+    condition_left_raw_variable = MDONG,
+    condition_operator = "if_checked",
+    sub_algorithm = hardcode_ct,
+    target_hardcoded_value = "ONGOING",
+    target_sdtm_variable_codelist_code = "C66728",
+    target_sdtm_var = CMENRTPT,
+    merge_to_topic_by = c(oak_id_vars,
+      topic_var_source = MDRAW
+    )
+  ) |>
+  # Derive qualifier CMSTRTPT  Annotation text = If checked then CM.CMENTPT = 'DATE OF LAST ASSESSMENT'
+  if_then_else(
+    raw_dataset = MD1,
+    raw_variable = MDONG,
+    condition_left_raw_dataset = MD1,
+    condition_left_raw_variable = MDONG,
+    condition_operator = "if_checked",
+    sub_algorithm = hardcode_no_ct,
+    target_hardcoded_value = "DATE OF LAST ASSESSMENT",
+    target_sdtm_var = CMENTPT,
+    merge_to_topic_by = c(oak_id_vars,
+      topic_var_source = MDRAW
+    )
+  ) |>
+  # Derive CMGRPID
+  assign_no_ct(
+    raw_dataset = MD1,
+    raw_variable = MDNUM,
+    target_sdtm_var = CMGRPID,
+    merge_to_topic_by = c(oak_id_vars,
+      topic_var_source = MDRAW
+    )
+  ) |>
+  dplyr::mutate(
+    STUDYID = "test_study",
+    DOMAIN = "CM",
+    CMCAT = "GENERAL CONMED"
+  ) |>
+  derive_usubjid() |>
+  derive_sequence(keys = c(USUBJID, CMTRT)) |>
+  derive_visit_visitnum() |>
+  derive_study_day(
```

Comment:
I see how this is related to my function `calculate_study_day` I was working on. So we will have another function `derive_study_day` as a wrapper or should I should modify my function to suite this purpose here?

Reply:
Updated the parameters to match `calculate_study_day`